pkg/controller/addon/addon_controller.go

```diff
@@ -140,10 +145,16 @@ func (r *ReconcileAddon) reconcileAddon(req reconcile.Request, res *op.Addon) (r
 		// handle error
 	}
 
-	// TODO: targetNamspace should be decided from current tekton pipelines installation
-	// if there is no pipeline installation then the default namespace 'tekton-pipelines'
-	// can be used
-	targetNamespace := "tekton-pipelines"
+	pipeline, err := r.getPipelineRes()
+	if err != nil {
+		_ = r.updateStatus(res, op.AddonCondition{
+			Code:    op.ErrorStatus,
+			Details: err.Error(),
+			Version: res.Spec.Version,
+		})
+		return reconcile.Result{}, err
+	}
+	targetNamespace := pipeline.Namespace
 	tfs := []mf.Transformer{
 		mf.InjectOwner(res),
 		mf.InjectNamespace(targetNamespace),
```

Comment:
Install `addon` to the same namespace of `pipeline`, so when CR `Config` deleted, does the namespace deleted?
What's the relationship between `Config` and `Addon`, what happened if `Config` deleted. 

hmm, does anyone think `Config` is a little confuse, could we make the name more make sense? like `TektonPipeline`

Reply:
Seems `Addon` set owner to `Config`...

Reply:
yes, addon owner is set to config.
so when pipeline is deleted, addons will be deleted.

> hmm, does anyone think Config is a little confuse, could we make the name more make sense? like TektonPipeline

we started with the name `install.operator.tekton.dev` instead of `config.operator.tekton.dev` :D
`config` is an upgrade compared to `install`
but, sure this is something we can discuss/decide :)